### PR TITLE
Raw api layer

### PR DESCRIPTION
**Pull request by bwbeach**
This is a proposal for a "raw API" layer underneath the B2Api class.  The idea is to have a super-simple HTTP wrapper class that doesn't need much testing itself, and can be replaced with a mock or simulator class for testing the layers of code above it.

### CODE REVIEW

File: b2 [hunk 1]
```diff
@@ -763,10 +763,75 @@ class AuthInfoCache(AbstractCache):
...
+    buckets and files returned are Python objects with accessor
+    methods.
+
+    Also,  keeps a cache of information needed to access the service,
+    such as auth tokens and upload URLs.
+    """
+
+    # TODO: move HTTP code out to B2RawApi
+    # TODO: ConsoleTool passes the account info cache into the constructor
+    # TODO: provide method to get the account info cache (so ConsoleTool can save it)
```
Comment (ppolewicz) [inline b2]: `B2Api.cache` is public and by default it uses StoredAccountInfo or wrapper over it, which saves the cache in `~/.b2_account_info`. Other (other than `ConsoleTool`) `B2Api` users are likely to use the same capability to just `api = B2Api()` and call methods on it, if they use a single thread.

File: b2 [hunk 2]
```diff
@@ -1327,6 +1390,17 @@ def download_file_by_name(args):
 
 
 class ConsoleTool(object):
+
+    """
+    Implements the commands available in the B2 command-line tool
+    using the B2Api library.
+
+    Responsible for storing cached account data in ~/.b2_account_info,
```
Comment (ppolewicz) [inline b2]: I don't think `ConsoleTool` should be responsible for that - other `B2Api` callers should be saved from doing it in their code unless it is necessary.

**Comment by ppolewicz** [comment]
I am a bit torn on this. We can mock urllib for tests as it currently is and splitting raw api from the public API adds a new layer, which makes working with the code slightly more complicated. It allows for more fine-grained testing, which we may not need at this point.

It will be a lot of work.

It will conflict with everything.

Lets do it anyway!

File: b2 (hunk 1 above)
Comment (bwbeach) [inline b2]: I think that's true, and it is fine as a default.  There may be users of the library, though, that don't want to use `~/.b2_account_info`, and want to use a different kind of cache.  They may want to provide the account ID and application key when they construct a B2Api, and then cache based on that.

File: b2 (hunk 2 above)
Comment (bwbeach) [inline b2]: How about:

> Uses the StoredAccountInfo object to keep account data in `.b2_account_info` between runs.

**Comment by bwbeach** [comment]
Will this conflict with any work you have in progress?  I'm happy to wait until an opportune time, and merge my changes then.

File: b2 (hunk 2 above)
Comment (ppolewicz) [inline b2]: Perfect

**Comment by ppolewicz** [comment]
It will conflict with nearly any change in `B2Api`. Please merge it as it currently is and lets sync when someone works on what so that we can avoid bigger conflicts.

File: b2 (hunk 1 above)
Comment (ppolewicz) [inline b2]: Indeed. They can pass `account_info` and `cache` arguments to `B2Api` for exactly that purpose. `InMemoryCache` is implemented already, but I haven't gotten around to `InMemoryAccountInfo` yet.

**Pull request merged**